### PR TITLE
Format currency per unit values like USD/kWh

### PR DESCRIPTION
**Pull request by felipecrs**
## Breaking change

N/A

## Proposed change

I started to play with energy measurement in my home recently, and since then I realized a few stack traces being printed in my browser console. Tracking down the issue, it seems that the unit of measurement that I used was not supported.

Specifically, I have RESTful sensor that collects the tariff from the energy provider. And the proper unit of measurement for the tariff is money per kilowatt hour.

So, this adds proper format for currencies with unit as unit of measurement, like USD/kWh.

Some demos, English:

chrome_9dUMqgeXV2

Portuguese BR:

chrome_akBxeAMzQ9

Previously they would all render as `99BRL/kWh`, and an error would be printed in the browser console that `BRL/kWh` isn't a supported currency.

Additionally, I spent some time improving the devcontainer experience.

Also, the debugger configuration wasn't working for me. I suppose something changed in the VSCode/the debugger/webpack since then. After spending some time on it, I realized that the default configuration for `webRoot` works while the one previously set here doesn't, so I simply removed it. Renamed `pwa-chrome` to `chrome` as the former was deprecated. I did not test the debug configuration for anything but Frontend (because the build task wasn't working for me and I didn't want to spend more time on it).

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
template:
  - sensor:
      - name: "Total cost"
        state: "99"
        device_class: monetary
        unit_of_measurement: "BRL"
        unique_id: total_cost
      - name: "Cost per kWh"
        state: "1"
        device_class: monetary
        unit_of_measurement: "BRL/kWh"
        unique_id: cost_per_kwh
```

## Additional information

- This PR fixes or closes issue: N/A
- This PR is related to issue or discussion: N/A
- Link to documentation pull request: N/A

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
the `device_class: monetary` can only have a [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217#Active_codes) UOM

**Comment by felipecrs** [comment]
So, this adds support for this extra notation.

**Comment by bramkragten** [comment]
Yeah I see that, but we don't support it. We can't just add support for these kind of things everywhere, as it will get very complicated to maintain and support in the future.

That's why for changes like this you will first need to open an architect discussion.

**Comment by felipecrs** [comment]
> everywhere

Is there anywhere else that needs to be changed to add support for this? Maybe the docs? I can handle that as well.

And this is not any corner case, the energy integration does allow you to have an entity for the tariff. This just adds proper support for its unit of measurement in Dashboard.

**Comment by felipecrs** [comment]
Just to say, I always hesitate to open an architect discussion. It seems so complicated for such a small thing.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -4,7 +4,25 @@
...
+    },
+    "8090": {
+      "label": "Demo",
+      "onAutoForward": "silent"
+    },
+    "8080": {
+      "label": "Cast",
+      "onAutoForward": "silent"
+    }
+  },
```
Comment (frenck) [inline .devcontainer/devcontainer.json:25]: This does not belong in this PR.

**Comment by bramkragten** [comment]
It is not just the frontend, but also core etc. Maybe we need an other device class for things like this? I have not thought it trough/went through all the code, so can't say atm...

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (felipecrs) [reply .devcontainer/devcontainer.json:25]: Why not? I had to change it in order to make the debugger work in the devcontainer. AppPort was causing VSCode to forward the Hass 8123 to 8124, while the debugger is configured to open on 8123.

With this change, VS Code will auto-select the best available port (8123 if available), but if it's not available, the debugger will also take the forwarded port into account and dynamically update it.

Comment (bramkragten) [reply .devcontainer/devcontainer.json:25]: The change doesn't have to be wrong, but a PR should change 1 thing, this should be in a separate PR so reviewing and merging is quicker and easier.

Comment (felipecrs) [reply .devcontainer/devcontainer.json:25]: Alright. Will do that. Thanks.

**Comment by felipecrs** [comment]
I can't imagine what has to be changed in the core. This basically only displays the data in a better way, this doesn't change the actual data.

> I have not thought it trough/went through all the code

Do you mean the code of this PR? It's quite simple. I just split the USD/kWh in two, format the value as the USD currency and add the `/kWh` suffix to the final text.
